schedulers/argo-workflow/addons.tf

```diff
@@ -0,0 +1,106 @@
+module "eks_blueprints_kubernetes_addons" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints//modules/kubernetes-addons?ref=v4.10.0"
+
+  eks_cluster_id       = module.eks_blueprints.eks_cluster_id
+  eks_cluster_endpoint = module.eks_blueprints.eks_cluster_endpoint
+  eks_oidc_provider    = module.eks_blueprints.oidc_provider
+  eks_cluster_version  = module.eks_blueprints.eks_cluster_version
+
+  #---------------------------------------------------------------
+  # Amazon EKS Managed Add-ons
+  #---------------------------------------------------------------
+  # EKS Addons
+  enable_amazon_eks_vpc_cni            = true
+  enable_amazon_eks_coredns            = true
+  enable_amazon_eks_kube_proxy         = true
+  enable_amazon_eks_aws_ebs_csi_driver = true
+
+  #---------------------------------------------------------------
+  # CoreDNS Autoscaler helps to scale for large EKS Clusters
+  #   Further tuning for CoreDNS is to leverage NodeLocal DNSCache -> https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/
+  #---------------------------------------------------------------
+  enable_coredns_autoscaler = true
+
+  #---------------------------------------------------------------
+  # Metrics Server
+  #---------------------------------------------------------------
+  enable_metrics_server = true
+
+
+  #---------------------------------------------------------------
+  # Cluster Autoscaler
+  #---------------------------------------------------------------
+  enable_cluster_autoscaler = true
+
+
+  #---------------------------------------------------------------
+  # Spark Operator Add-on
+  #---------------------------------------------------------------
+  enable_spark_k8s_operator = true
+
+  #---------------------------------------------------------------
+  # Apache YuniKorn Add-on
+  #---------------------------------------------------------------
+  enable_yunikorn = true
+
+
+  depends_on = [
+    module.eks_blueprints
+  ]
```

Comment:
This dependency causes issues to Add-ons module by replacing(recreating) some of the resources(mainly Addons IRSA). We have seen this before in EKS Blueprints and we removed this from all our patterns